overlord/snapstate/handlers.go

```diff
@@ -818,6 +852,20 @@ func (m *SnapManager) undoUnlinkCurrentSnap(t *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
+	// Restore disabled services
+	var missingSvcs []string
+	missingSvcs, err = m.backend.RestoreDisabledServices(oldInfo, snapst.LastActiveDisabledServices, progress.Null)
+	if err != nil {
+		// nothing to undo, since worst case we would have just disabled some
+		// services which will now be removed anyways
+		return err
```

Comment:
Perhaps you could log the error in the task too

Reply:
Okay, added